packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -34,21 +46,67 @@ public function __construct(string $definitionDirectory, string $indexPrefix, Cl
         $this->client = $client;
     }
 
+    /**
+     * @param int $buildVersion
+     * @internal Will be replaced with constructor injection in the next major release
+     */
+    public function setBuildVersion(int $buildVersion): void
+    {
+        if ($this->buildVersion !== null) {
+            throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
+        }
+
+        $this->buildVersion = $buildVersion;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Setting\Setting $setting
+     * @internal Will be replaced with constructor injection in the next major release
+     */
+    public function setSetting(Setting $setting): void
+    {
+        if ($this->setting !== null) {
+            throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
+        }
+
+        $this->setting = $setting;
+    }
+
     /**
      * @param int $domainId
      * @param string $index
      * @return string
      */
     public function getIndexName(int $domainId, string $index): string
     {
-        return $this->indexPrefix . $index . $domainId;
+        return $this->indexPrefix . $this->buildVersion . $index . $domainId;
```

Comment:
you are using a different name of an index, it's like a different database name. It's a BC break in my opinion. The project simply won't work after `composer update`, am I right?

Reply:
no, it isn't BC break. You have to run a phing target for correct elastic data, but code is BC